aya/src/maps/mod.rs

```diff
@@ -304,6 +317,45 @@ impl Map {
         Ok(fd)
     }
 
+    /// Loads a map from a pinned path in bpffs.
+    pub fn from_pinned<P: AsRef<Path>>(path: P) -> Result<Map, MapError> {
+        let path_string = CString::new(path.as_ref().to_str().unwrap()).unwrap();
+        let fd =
+            bpf_get_object(&path_string).map_err(|(code, io_error)| MapError::SyscallError {
+                call: "BPF_OBJ_GET".to_owned(),
+                code,
+                io_error,
+            })? as RawFd;
+
+        let info = bpf_map_get_info_by_fd(fd).map_err(|io_error| MapError::GetMapFDError {
+            call: "BPF_OBJ_GET_INFO_BY_FD".to_owned(),
+            io_error,
+        })?;
+
+        Ok(Map {
+            obj: obj::parse_map_info(info, true),
+            fd: Some(fd),
+            btf_fd: None,
+            pinned: true,
+        })
+    }
+
+    /// Loads a map from a rawfd.
+    /// We assume that the map is not pinned in this case.
```

Comment:
I think we should elaborate on the use case a little here to avoid confusion.

```rust
/// Loads a map from a [`RawFd`].
///
/// If loading from a BPF Filesystem (bpffs) you should use [`Map::from_pinned`].
/// This API is intended for cases where you have received a valid BPF FD from some other means.
/// For example, you received an FD over Unix Domain Socket.
```

Reply:
Done